src/Microsoft.AspNetCore.Sockets/Transports/ServerSentEventsTransport.cs

```diff
@@ -24,27 +25,43 @@ public ServerSentEventsTransport(ReadableChannel<Message> application, ILoggerFa
         public async Task ProcessRequestAsync(HttpContext context)
         {
             context.Response.ContentType = "text/event-stream";
+
+            // Working around dynamic compression behavior in ANCM: https://github.com/aspnet/AspNetCoreModule/issues/16
             context.Response.Headers["Cache-Control"] = "no-cache";
             context.Response.Headers["Content-Encoding"] = "identity";
+
             await context.Response.Body.FlushAsync();
 
             try
             {
                 while (await _application.WaitToReadAsync(context.RequestAborted))
                 {
                     Message message;
+                    var counter = 0;
                     while (_application.TryRead(out message))
                     {
+                        counter++;
                         using (message)
                         {
                             await Send(context, message);
                         }
                     }
+
+                    _logger.LogTrace("Sent batch of {0} frames", counter);
                 }
+
+                // If the completion is faulted or cancelled, we want to manifest that error
+                _application.Completion.GetAwaiter().GetResult();
```

Comment:
Same as above, I need to understand why this is important. What pieces of the system need to be notified of the exception and how does that flow.